### PR TITLE
[sailfish-browser] Add ability of calculating cache and site data size. JB#56152

### PR DESCRIPTION
**Pull request by atatarov**

### CODE REVIEW

**Comment by atatarov** [comment]
It depends on https://github.com/sailfishos/embedlite-components/pull/70/files

**Review by okodron — APPROVED** [review]

File: apps/core/settingmanager.cpp [hunk 1]
```diff
@@ -90,6 +93,22 @@ void SettingManager::removeAllTabs()
     DBManager::instance()->removeAllTabs();
 }
 
+void SettingManager::calculateCacheSize(QJSValue callback)
+{
+    if (!callback.isNull() && !callback.isUndefined() && callback.isCallable()) {
+        m_calculateCacheSizeCb.reset(new QJSValue(callback));
+    }
+    SailfishOS::WebEngine::instance()->notifyObservers(QString("get-cache-size"), QVariant());
```
Comment (rainemak) [inline apps/core/settingmanager.cpp:101]: This notify should be inside the if-block as reported cache size is only handled when there is a callback.

File: apps/core/settingmanager.cpp [hunk 2]
```diff
@@ -90,6 +93,22 @@ void SettingManager::removeAllTabs()
...
+}
+
+void SettingManager::calculateSiteDataSize(QJSValue callback)
+{
+    if (!callback.isNull() && !callback.isUndefined() && callback.isCallable()) {
+        m_calculateSiteDataSizeCb.reset(new QJSValue(callback));
+    }
+    SailfishOS::WebEngine::instance()->notifyObservers(QString("get-site-data-size"), QVariant());
```
Comment (rainemak) [inline apps/core/settingmanager.cpp:109]: Same as above

**Comment by rainemak** [comment]
Please rebase as well.

**Review by rainemak — APPROVED** [review]

**Pull request merged**